Sources/TuistCore/Utils/XcodeBuildLocator.swift

```diff
@@ -0,0 +1,77 @@
+import Foundation
+import TSCBasic
+import TuistGraph
+import TuistSupport
+
+public protocol XcodeBuildLocating {
+
+    /// Locates the build output directory for `xcodebuild` command.
+    ///
+    /// - Parameters:
+    ///   - platform: The platform for the built scheme.
+    ///   - projectPath: The path of the Xcode project or workspace.
+    ///   - configuration: The configuration name, i.e. `Release`, `Debug`, or something custom.
+    func locateBuildDirectory(
+        platform: Platform,
+        projectPath: AbsolutePath,
+        configuration: String
+    ) throws -> AbsolutePath
+
+    /// Locates the products output directory for `xcodebuild` command.
+    ///
+    /// - Parameters:
+    ///   - projectPath: The path of the Xcode project or workspace.
+    func locateProductsDirectory(for projectPath: AbsolutePath) throws -> AbsolutePath
```

Comment:
If we rename the class, its name will already be explicit about what we are locating so we can rename this one to just `locate` and remove some naming redundancy.